### PR TITLE
Docs: lightweight doc autogen

### PR DESCRIPTION
**Pull request by skmp**
This generates a doc outline from tags in the .c/.cpp/.cc/.h/.hpp files.

To be used at least between releases to update the doc file

To Generate, run in git root folder ```Scripts/generate_doc_outline.sh > docs/SourceOutline.md```

Tags look like
```
$info$
glossary: <name> ~ <definition> (Optional, registers or replaces a glossary entry)
glossary: IR ~ Intermidiate Representation, a of storage for our high-level opcode representation
glossary: SSA ~ Single Static Assignment, a form of representing IR in memory
glossary: Basic Block ~ A block of instructions with no control flow, terminated by control flow
glossary: Fragment ~ A Collection of basic blocks, possible an entire guest function or a fraction of it
category: <name> ~ <description> (Optional, registers or replaces a category description)
category: backend ~ Concerns itself with generating binary code from (optimized) IR
meta: <name> ~ <description> (Optional, registers or replaces a meta, aka tag, description)
meta: backend|arm64 ~ Arm64 Splatter backend
tags: <meta name> [, <meta name>, ...] (Required if info tag exists)
tags: backend|arm64
desc: <short file description> (Optional)
desc: Main glue logic of the arm64 splatter backend
$end_info$
```

Once the files are annotated, it [generates MD like this](https://github.com/FEX-Emu/FEX/blob/skmp/lightweight-docs-3/docs/SourceOutline.md)

### CODE REVIEW

File: Scripts/generate_release.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+echo "Tagging FEX-$2, previous release FEX-$1"
+echo "Press cltr-c to cancel within 10 seconds"
+sleep 10
+
+git tag FEX-$2 -a -m "temporary"
+Scripts/generate_doc_outline.sh > docs/SourceOutline.md
+git commit docs/SourceOutline.md -m "Docs: Update for release FEX-$2"
+git tad -d FEX-$2
+git tag FEX-2104 -a -m "$(Scripts/generate_changelog.sh FEX-$1 FEX-$2)" --edit
```
Comment (Sonicadvance1) [inline Scripts/generate_release.sh:18]: Looks like this shouldn't be `FEX-2104` on this line

File: docs/SourceOutline.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# FEXCore
+
+## Glossary
+
+- Splatter: a code generator backend that concaternates configurable macros instead of doing isel
+- IR: Intermidiate Representation, a of storage for our high-level opcode representation
```
Comment (Sonicadvance1) [inline docs/SourceOutline.md:6]: intermediate

File: docs/SourceOutline.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- [MiscOps.cpp](../External/FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp): Misc ops of the arm64 splatter backend
+- [MoveOps.cpp](../External/FEXCore/Source/Interface/Core/JIT/Arm64/MoveOps.cpp): Move ops of the arm64 splatter backend
+- [VectorOps.cpp](../External/FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp): Vector ops of the arm64 splatter backend
+
+
+
+## frontend
+
+### x86-meta-blocks
+- [Frontend.cpp](../External/FEXCore/Source/Interface/Core/Frontend.cpp): Extracts instruction & block meta info, frontend mutliblock logic
```
Comment (Sonicadvance1) [inline docs/SourceOutline.md:33]: multiblock

File: Scripts/generate_release.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/env bash
```
Comment (Sonicadvance1) [inline Scripts/generate_release.sh:1]: As the effective release manager I'll probably be modifying this file after this PR merge.
Good to have as the initial example still though

**Review by Sonicadvance1 — COMMENTED** [review]
Looks pretty good aside from the few comments that I had

File: Scripts/generate_release.sh (hunk 1 above)
Comment (skmp) [reply Scripts/generate_release.sh:18]: Woops

File: Scripts/generate_release.sh (hunk 4 above)
Comment (skmp) [reply Scripts/generate_release.sh:1]: Sounds good

File: docs/SourceOutline.md (hunk 3 above)
Comment (skmp) [reply docs/SourceOutline.md]: Fixed, doc tags now live in #917

File: docs/SourceOutline.md (hunk 2 above)
Comment (skmp) [reply docs/SourceOutline.md]: Fixed, doc tags now live in #917

**Review by Sonicadvance1 — APPROVED** [review]

**Pull request merged**